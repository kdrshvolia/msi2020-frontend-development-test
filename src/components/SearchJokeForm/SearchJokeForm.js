import React from 'react';import Radio from '../Radio/Radio';import Button from '../Button/Button';const SearchJokeForm = ({ setType, getJoke }) => {  return (    <div>      <Radio name="search-type" value="Random" handleChange={setType} />      <Radio name="search-type" value="From categories" handleChange={setType} />      <Radio name="search-type" value="Search" handleChange={setType} />      <Button text="Get a joke" onClick={getJoke} />    </div>  );};export default SearchJokeForm;