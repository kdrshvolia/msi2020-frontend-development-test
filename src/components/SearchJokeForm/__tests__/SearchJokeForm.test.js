import React from 'react';import renderer from 'react-test-renderer';import SearchJokeForm from '../SearchJokeForm';describe('SearchJokeForm', () => {  it('Reders correctly', () => {    const categories = ['a', 'b', 'c'];    const options = { a: 'a', b: 'b' };    const getJoke = jest.fn();    const setCategory = jest.fn();    const rendered = renderer.create(      <SearchJokeForm        categories={categories}        options={options}        getJoke={getJoke}        setCategory={setCategory}      />,    );    expect(rendered).toMatchSnapshot();  });});