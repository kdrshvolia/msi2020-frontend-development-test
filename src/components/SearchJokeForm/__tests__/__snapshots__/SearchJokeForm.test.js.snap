// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchJokeForm Reders correctly 1`] = `
<div>
  <div
    className="sc-AxjAm dWVCDA"
  >
    <input
      checked={true}
      className="sc-AxirZ hxQKcv"
      id="Random"
      name="search-type"
      onChange={[Function]}
      type="radio"
    />
     
    <label
      htmlFor="Random"
    >
      Random
    </label>
  </div>
  <div
    className="sc-AxjAm dWVCDA"
  >
    <input
      checked={true}
      className="sc-AxirZ hxQKcv"
      id="From categories"
      name="search-type"
      onChange={[Function]}
      type="radio"
    />
     
    <label
      htmlFor="From categories"
    >
      From categories
    </label>
  </div>
  <div
    className="sc-AxhCb gGkwKU"
  >
    <div
      className="sc-AxhUy jPMDTT"
      onClick={[Function]}
    >
      A
    </div>
    <div
      className="sc-AxhUy jPMDTT"
      onClick={[Function]}
    >
      B
    </div>
    <div
      className="sc-AxhUy jPMDTT"
      onClick={[Function]}
    >
      C
    </div>
  </div>
  <div
    className="sc-AxjAm dWVCDA"
  >
    <input
      checked={true}
      className="sc-AxirZ hxQKcv"
      id="Search"
      name="search-type"
      onChange={[Function]}
      type="radio"
    />
     
    <label
      htmlFor="Search"
    >
      Search
    </label>
  </div>
  <div
    className="sc-AxgMl hTOpWV"
  >
    <input
      className="sc-AxheI EDBAb"
      onChange={[Function]}
      placeholder="Free text search..."
      type="text"
    />
  </div>
  <div>
    <button
      className="sc-AxiKw bNMzOj"
      onClick={[Function]}
      type="button"
    >
      Get a joke
    </button>
  </div>
</div>
`;
